diff --git a/dist/Toast.svelte b/dist/Toast.svelte
index f1088e3fe129d167d01eb90c921f0b498f65ec13..74234bc2ab8ef4275458a031b781c7b574bd103e 100644
--- a/dist/Toast.svelte
+++ b/dist/Toast.svelte
@@ -320,12 +320,7 @@ const icon = $derived.by(() => {
 
 {#snippet LoadingIcon()}
 	{#if loadingIcon}
-		<div
-			class={cn(classes?.loader, toast?.classes?.loader, 'sonner-loader')}
-			data-visible={toastType === 'loading'}
-		>
-			{@render loadingIcon()}
-		</div>
+		{@render loadingIcon()}
 	{:else}
 		<Loader
 			class={cn(classes?.loader, toast.classes?.loader)}
